### PR TITLE
Security Definition Updates

### PR DESCRIPTION
**Pull request by john-arccos**
# Commit Message

* creating new parameter `useRedirectUI` that enables hosting an endpoint for using the standard OAuth 2.0 Authorization Code Grant Flow.

* Updating swagger plugin to include middleware that puts scopes next to the authorization widget for each endpoint

# Summary

Love the package you've put together, but wanted to make a couple modifications to better support authentication for my use case. I made the changes listed in the commits above and also just a couple minor clean up things around the code base. You can see my hosted version [here](https://api.arccosgolf.com/swagger), but you I can't give you credentials (work system). Let me know what you think!

# Example

Here's a screenshot of what the scope definition additions looks like:
Screen Shot 2022-11-21 at 12 43 58 PM

Here's the swagger file I'm including in  to accomplish this:
```{
  "swagger": "2.0",
  "securityDefinitions": {
    "AccessCodeAuth": {
      "type": "oauth2",
      "flow": "accessCode",
      "authorizationUrl": "https://signin.arccosgolf.com/login",
      "tokenUrl": "https://signin.arccosgolf.com/oauth2/token",
      "oauth2RedirectUrl": "https://api.arccosgolf.com/swagger",
      "scopes": {
        "openid": "Required for all endpoints that have userId in the path",
        "arccos/read:rounds": "Grants read rounds access",
        "arccos/read:users": "Grants read users access"
      }
    }
  }
}
```

and here's an example `httpApi` `security` attribute I'm using:
```security: [
    {
        AccessCodeAuth: [
            `arccos/read:rounds`,
        ],
    },
],
```

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -27,6 +34,7 @@ module.exports = {
     '@typescript-eslint/explicit-function-return-type': 0,
     'deprecation/deprecation': 1,
     '@typescript-eslint/no-explicit-any': 0, // Ideally we want this on, but we have some types that I'm unfamiliar with typed as `any`.
+    '@typescript-eslint/no-unused-vars': 0,
```
Comment (bfaulk96) [inline .eslintrc.js:37]: Any reason we are turning off the Typescript check in favor of the JS check?

**Comment by bfaulk96** [comment]
I'll try to take a deeper look into this over the weekend if I can find some time, I apologize!

File: .eslintrc.js (hunk 1 above)
Comment (john-arccos) [reply .eslintrc.js:37]: I just wasn't sure if I could configure it the same way as the default one. I can change it if you want

**Comment by john-arccos** [comment]
verified end result is unchanged (still working) after my last commit

**Comment by bfaulk96** [comment]
Got a chance to look at this fully, looks good. Thanks for your contribution! I'll release in 2.12.0.

**Pull request merged**